src/common/transformations/include/transformations/utils/utils.hpp

```diff
@@ -236,12 +236,14 @@ TRANSFORMATIONS_API std::vector<Input<Node>> get_node_target_inputs(const std::s
 TRANSFORMATIONS_API std::shared_ptr<Node> node_to_get_shape_value_of_indices_from_shape_node(
     const std::shared_ptr<Node>& shape_node,
     const std::vector<size_t>& indices,
-    const std::vector<std::shared_ptr<Node>>& copy_rt_info_from = {});
+    const std::vector<std::shared_ptr<Node>>& copy_rt_info_from = {},
+    const ov::element::Type& shape_path_precision = ov::element::i64);
```

Comment:
May be I missed something, but I didn't see how this new parameter is used in this PR. Do we really need this change?

Reply:
`node_to_get_shape_value_of_indices_from_shape_node` is used in function `node_to_get_shape_value_of_indices_from_shape_source`, which in turn is used to create shapeof subgraphs in [merge_transpose_reorder.cpp cpu tests](https://github.com/openvinotoolkit/openvino/pull/18999/files?diff=split&w=0#diff-dc6f3e5783581766a8831fdf0309aecc32f9762b69fad8d2af8567da0185ad32R104). I added `shape_path_precision` parameter to create the shapeof subgraph in i32 precision and avoid unnecessary conversion i64->i32 converstion.